src/poetry/utils/authenticator.py

```diff
@@ -91,7 +91,12 @@ class AuthenticatorRepositoryConfig:
 
     def __post_init__(self) -> None:
         parsed_url = urllib.parse.urlsplit(self.url)
-        self.netloc = parsed_url.netloc
+        # Remove username:password from netloc string
+        if "@" in parsed_url.netloc:
+            self.netloc = parsed_url.netloc.rsplit("@", 1)[-1]
+        else:
+            self.netloc = parsed_url.netloc
+
```

Comment:
```suggestion
        self.netloc = parsed_url.netloc
```
I think that in order to ensure the consistency of netloc you shouldn't remove the login info from it.